### PR TITLE
TW-1031: Improve Emoji Picker / Input Row

### PR DESCRIPTION
**Pull request by imGok**
- Keep input focus when clicking on emojis, emoji button
- Close on sending
- Draft & Chat use the same input row now 

# Chat

https://github.com/linagora/twake-on-matrix/assets/48354990/4767a9a2-54d8-49eb-b093-ac21d03c5606

https://github.com/linagora/twake-on-matrix/assets/48354990/4e6364e0-65f7-43c6-9148-91717c3e0865

# Draft Chat

https://github.com/linagora/twake-on-matrix/assets/48354990/8beada39-502c-4f7a-a8c0-d69d0debe090

https://github.com/linagora/twake-on-matrix/assets/48354990/8e3611c8-6b82-4217-8a92-981316954330

#1031

### CODE REVIEW

**Comment by hoangdat** [comment]
I tried, but after send message, `focus` also lost in composer. Hope we can keep it in composer.

**Comment by imGok** [comment]
Fixed

https://github.com/linagora/twake-on-matrix/assets/48354990/0def1ed2-6f92-43a4-b300-cbd5b1d18457

**Comment by sherlockvn** [comment]
image
When i change emoji group, please make it keep focus (currently i can't enter after i change group of emoji)

**Comment by imGok** [comment]
> image When i change emoji group, please make it keep focus (currently i can't enter after i change group of emoji)

@sherlockvn current lib doesn't allow us to play with category click event, we would need to do our own custom widget to achieve what you want

**Comment by hoangdat** [comment]
> When i change emoji group, please make it keep focus (currently i can't enter after i change group of emoji)

IMO, it is good now, after `change group` -> `pick other emoji` -> focus come back.

File: lib/pages/chat/chat.dart
Comment (hoangdat) [inline lib/pages/chat/chat.dart:1]: - [ ] should apply the same for `draft_chat`

**Comment by sherlockvn** [comment]
Please make the demo in draft chat and mobile also

File: lib/pages/chat/chat.dart
Comment (imGok) [reply lib/pages/chat/chat.dart:1]: I refactored a lot of code in my last commit in order to have the same input row as chat one, please check again

**Comment by sherlockvn** [comment]
The last demo when you switch to stickers and icons, the height of its so high?. 
Screenshot 2024-02-02 at 15 16 58

**Comment by imGok** [comment]
> The last demo when you switch to stickers and icons, the height of its so high?. Screenshot 2024-02-02 at 15 16 58

Closing keyboard and showing emoji are not at the same time so with a laggy device it will behave like this

**Comment by hoangdat** [comment]
Hi Julian, I dont know where we have side effect but https://github.com/linagora/twake-on-matrix/issues/1450, please fix it also

**Comment by imGok** [comment]
> Hi Julian, I dont know where we have side effect but #1450, please fix it also

Fixed in my last commit https://github.com/linagora/twake-on-matrix/pull/1413/commits/63bb8f50dd00616e5533b09d8a39bacbcdd87728

https://github.com/linagora/twake-on-matrix/assets/48354990/402c85ad-a30e-4027-a9ac-6399fd8e343d

**Comment by sherlockvn** [comment]
### Latest demo:

https://github.com/linagora/twake-on-matrix/assets/43041967/ffe88f86-9487-4a07-b26d-133b0e885c1f

File: lib/pages/chat/chat_emoji_picker.dart [hunk 1]
```diff
@@ -1,26 +1,41 @@
...
       builder: (context, showEmojiPicker, _) {
         return AnimatedContainer(
+          decoration: const BoxDecoration(
+            border: Border(
+              top: BorderSide(
+                color: Colors.transparent,
+                width: 1,
+              ),
+            ),
+          ),
```
Comment (nqhhdev) [inline lib/pages/chat/chat_emoji_picker.dart:30]: Why we need it with color is transparent?

File: lib/pages/chat_draft/draft_chat.dart [hunk 2]
```diff
@@ -79,16 +80,19 @@ class DraftChatController extends State<DraftChat>
 
   bool showScrollDownButton = false;
 
-  String inputText = '';
+  ValueNotifier<String> inputText = ValueNotifier('');
```
Comment (nqhhdev) [inline lib/pages/chat_draft/draft_chat.dart:83]: Don't forget dispose it

File: lib/pages/chat_draft/draft_chat.dart [hunk 3]
```diff
@@ -79,16 +80,19 @@ class DraftChatController extends State<DraftChat>
...
 
-  bool showEmojiPicker = false;
+  ValueNotifier<bool> showEmojiPickerNotifier = ValueNotifier(false);
```
Comment (nqhhdev) [inline lib/pages/chat_draft/draft_chat.dart:85]: same

File: lib/pages/chat/chat_emoji_picker.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/chat/chat_emoji_picker.dart:30]: no need the divider between the composer and emoji picker, if have, it looks ugly when run the animation

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**